conans/client/graph/range_resolver.py

```diff
@@ -69,13 +70,14 @@ def resolve(self, require, base_conanref, update, remote_name):
     def _resolve_local(self, search_ref, version_range):
         local_found = search_recipes(self._client_cache, search_ref)
         if local_found:
-            resolved_version = self._resolve_version(version_range, local_found)
-            if resolved_version:
-                return resolved_version
+            return self._resolve_version(version_range, local_found)
 
     def _resolve_remote(self, search_ref, version_range, remote_name):
         # We should use ignorecase=False, we want the exact case!
-        remote_found = self._remote_search.search_remotes(search_ref, remote_name)
+        remote_found = self._cached_remote_found.get(search_ref)
```

Comment:
Two questions:

- Is it not needed to store the remote in the cache?
- Is it not recommended to store the "None" or something when it is not found to not search again?

Reply:
- Good catch. It cannot change between conan invocations, that is the reason I didn't cache per remote. But via the API it can. So I will cache per remote.
- ``None`` is the value used for not having it in the cache. Should be other value. I thought, ``remote_found = self._remote_search.search_remotes(search_ref, remote_name)`` would always return at least an empty list, but as it is possible that it return None, I will cache another value.

Good catches.